spec/abstract-jobs.html

```diff
@@ -111,12 +111,24 @@ <h1>Execution</h1>
 
       <ul>
         <li>
-          If all references to a ECMAScript object are from a
-          [[Target]] field or internal slot, including the agent's [[KeptAlive]] field,
-          the implementation may replace the reference to the object in all such
-          [[Target]] fields or internal slots with the value ~empty~.
+          If the following conditions about an ECMAScript object are met, then
+          the implementation may replace each reference to the object from the
+          [[Target]] field of a FinalizationGroup's [[Cells]] List, or the
+          [[Target]] internal slot of a WeakRef, to the value ~empty~:
+          <ul>
+            <li>The object is not included in any agent's [[KeptAlive]] List.</li>
+            <li>
+                No possible future execution of the program would observably
+                reference the object, except through a chain of references which
+                includes a [[Target]] field or [[Target]] internal slot.
```

Comment:
this, i think, still needs the same clarification in lines 116-117? (an alternative would be an abstract operation rather than using prose to describe [[Target]] interaction)

Reply:
Well, I made sure [[Target]] referred separately to "field" and "internal slot"; isn't it pretty clear what we're talking about given the mention a couple lines earlier? Nothing else uses an internal slot of this name.

Reply:
I’m thinking about future uses which might, but might not be intended to be included here. It seems like it wouldn’t hurt to be overly explicit here, even though it’s certainly implied?

Reply:
I don't think that's how the ES spec is generally written. If an internal slot name is accidentally reused, then, for example, type checks will accidentally pass. The name "[[Target]]" is solidly owned by this proposal, or it will break.

Reply:
Anyway, fixed it.

Reply:
Fair point - the updated version seems good to me, thanks!